src/main/integrations/sentry-minidump/minidump-parser.ts

```diff
@@ -0,0 +1,256 @@
+// This code takes a lot of inspiration and info from the rust-minidump crate at this point in time:
+// https://github.com/rust-minidump/rust-minidump/tree/760f84058b909ff0f980988bd09f3a0f0421d298
+//
+// rust-minidump has the following license:
+//
+// MIT License
+//
+// Copyright (c) 2015-2023 rust-minidump contributors
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+export const MINIDUMP_MAGIC_SIGNATURE = 'MDMP';
+
+// https://docs.microsoft.com/en-us/windows/win32/api/minidumpapiset/ns-minidumpapiset-minidump_header
+type MinidumpHeader = {
+  signature: string;
+  version: number;
+  streamCount: number;
+  streamDirectoryRva: number;
+  checksum: number;
+  timeDateStamp: Date;
+  flags: bigint;
+};
+
+function readHeader(buf: Buffer): MinidumpHeader {
+  // pub struct MINIDUMP_HEADER {
+  //   pub signature: u32,
+  //   pub version: u32,
+  //   pub stream_count: u32,
+  //   pub stream_directory_rva: RVA,
+  //   pub checksum: u32,
+  //   pub time_date_stamp: u32,
+  //   pub flags: u64,
+  // }
+
+  return {
+    signature: buf.subarray(0, 4).toString(),
+    version: buf.readUInt32LE(4),
+    streamCount: buf.readUInt32LE(8),
+    streamDirectoryRva: buf.readUInt32LE(12),
+    checksum: buf.readUInt32LE(16),
+    timeDateStamp: new Date(buf.readUInt32LE(20) * 1000),
+    flags: buf.readBigUInt64LE(24),
+  };
+}
+
+type Location = {
+  dataSize: number;
+  rva: number;
+};
+
+// https://docs.microsoft.com/en-us/windows/win32/api/minidumpapiset/ns-minidumpapiset-minidump_location_descriptor
+function readLocationDescriptor(buf: Buffer, base: number): Location {
+  // pub struct MINIDUMP_LOCATION_DESCRIPTOR {
```

Comment:
Let's move this to where the type is being defined above, and add comments about what the field means.

Reply:
It was here to document the sizes of the fields for the parsing just below it. If it gets moved to the type definition it's just documenting the fields have the same names?

Reply:
I'd rather this be with the type itself, but given this is basically a constructor I see the value in doing it this way. We don't need to change it.

Reply:
I'll have another go and copy the rust docs across for the properties because they're useful. I'll document the offsets property rather than just copy the rust structs 🙂